src/mgmt/srv_pool.c

```diff
@@ -659,16 +687,68 @@ ds_mgmt_pool_list(void)
 
 	D_DEBUG(DB_MGMT, "pools:\n");
 	rc = rdb_tx_iterate(&tx, &svc->ms_pools, false /* !backward */,
-			    enum_pool_cb, NULL /* arg */);
+			    enum_pool_cb, &iter_args);
 
 	ABT_rwlock_unlock(svc->ms_lock);
 	rdb_tx_end(&tx);
+
 out_svc:
 	ds_mgmt_svc_put_leader(svc);
 out:
+	if (rc == 0) {
+		*npools = iter_args.npools;
+		*max_nsvc = iter_args.max_nsvc;
+		*poolsp = iter_args.pools;
+		*pools_len = iter_args.pools_index;
+	} else {
+		if (iter_args.pools) {
+			int pc;
+			for(pc = 0; pc < iter_args.pools_index; pc++)
```

Comment:
(style)  Missing a blank line after declarations
(style)  space required before the open parenthesis '('